### PR TITLE
Refactor service management codes

### PR DESCRIPTION
**Pull request by liangxin1300**
## Problem
1. Some service management related functions were here and there, and some of them were redundant
2. It will more convenient to make all of these functions running remotely or locally
## Solution
1. Create class `ServerManager` which has a set of service management functions
2. Create shortcuts for the classmethod to make sure the API wouldn't change

This PR would be small enough to backport to other branches

### CODE REVIEW

File: crmsh/utils.py [hunk 1]
```diff
@@ -2512,34 +2473,141 @@ def check_file_content_included(source_file, target_file):
...
+            rc, output, err = get_stdout_stderr(cmd)
+        if rc != 0 and err:
+            raise ValueError("Run \"{}\" error: {}".format(cmd, err))
+        return rc == 0, output
+
+    def is_available(self):
+        return self.service_name in self._do_action("list-unit-files")[1]
+
+    def is_enabled(self):
+        return self._do_action("is-enabled")[0]
```
Comment (arbulu89) [inline crmsh/utils.py:2510]: I guess we should use the strings from `ACTION_LIST`, right?
Instead of duplicating here

File: crmsh/utils.py [hunk 2]
```diff
@@ -2512,34 +2473,141 @@ def check_file_content_included(source_file, target_file):
...
+    @classmethod
+    def disable_service(cls, name, remote_addr=None):
+        """
+        Disable service
+        """
+        inst = cls(name, remote_addr)
+        inst.disable()
+
+    @classmethod
+    def enable_start_service(cls, name, remote_addr=None):
```
Comment (arbulu89) [inline crmsh/utils.py:2584]: This name looks pretty strange.
I would only have `start_service` and add a parameter named `enabled=False`, or something like that. And if this parameters comes, just run the `enable` method.
What do you think?

File: crmsh/utils.py [hunk 3]
```diff
@@ -2512,34 +2473,141 @@ def check_file_content_included(source_file, target_file):
...
+    def disable_stop_service(cls, name, remote_addr=None):
+        """
+        Disable and stop service
+        """
+        inst = cls(name, remote_addr)
+        inst.disable()
+        inst.stop()
+
+
+service_is_available = ServiceManager.service_is_available
```
Comment (arbulu89) [inline crmsh/utils.py:2602]: This looks a bit strange, but well. It's correct anyway

**Review by arbulu89 — CHANGES_REQUESTED** [review]
@liangxin1300 In general the changes are good.
I suggest to use the service action names from the already created constant, and maybe re-do the the `enable_start_service` and `disable_stop_service`

File: crmsh/utils.py (hunk 1 above)
Comment (liangxin1300) [reply crmsh/utils.py]: OK, changed

File: crmsh/utils.py (hunk 2 above)
Comment (liangxin1300) [reply crmsh/utils.py]: Ok, changed as `enable` parameter in `start_service` function, and `disable` parameter in `stop_service` function

**Comment by liangxin1300** [comment]
@arbulu89 Already changed! Thanks!

File: crmsh/utils.py [hunk 4]
```diff
@@ -2512,34 +2473,125 @@ def check_file_content_included(source_file, target_file):
...
+            prompt_msg = "Run \"{}\" on {}".format(cmd, self.remote_addr)
+            rc, output, err = run_cmd_on_remote(cmd, self.remote_addr, prompt_msg)
+        else:
+            rc, output, err = get_stdout_stderr(cmd)
+        if rc != 0 and err:
+            raise ValueError("Run \"{}\" error: {}".format(cmd, err))
+        return rc == 0, output
+
+    def is_available(self):
+        return self.service_name in self._do_action(self.ACTION_LIST[-1])[1]
```
Comment (arbulu89) [inline crmsh/utils.py:2507]: This looks pretty ugly (`self.ACTION_LIST[-1]`). I would change the `ACTION_LIST` from being a list to be a map or named list. The code would be neater.

**Review by arbulu89 — APPROVED** [review]
@liangxin1300 Approved.
I still see room from improvement changing how the `ACTION_LIST` works.

**Pull request merged**